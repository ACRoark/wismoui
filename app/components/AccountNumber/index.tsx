import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
