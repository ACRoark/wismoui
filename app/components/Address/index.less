@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
