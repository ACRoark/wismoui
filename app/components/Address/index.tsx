import React from 'react';

import AddressFormat from 'components/Address/AddressFormat';
import IAddress from 'types/IAddress';

import formatAddress from './formatAddress';

import './index.less';

interface IAddressProps {
  address: IAddress;
  format?: AddressFormat;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { address, format = 'block' } = props;

  return <address className="dte-wismo-address">{formatAddress(address, format)}</address>;
};

export default Address;
