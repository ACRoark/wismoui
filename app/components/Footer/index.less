@import (reference) '../../styles/themes/index.less';

@footer-cls-prefix: ~'@{dte-prefix}-footer';

.@{footer-cls-prefix} {
  align-items: center;
  background-color: @dte-background-brand-blue;
  bottom: 0;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: @footer-height; // Footer height must match <Content /> padding-bottom so the footer doesn't overlap the content
  justify-content: center;
  position: absolute;
  width: 100%;

  @media @mobile {
    height: @footer-mobile-height; // Footer height must match <Content /> padding-bottom
  }

  &-copyright {
    color: @dte-content-primary-inverse;
    display: flex;
    flex-wrap: wrap;
    font-size: @dte-wismo-font-size-xs;
    padding-top: 40px;
  }

  &-content {
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
    width: 100%;

    @media @mobile {
      padding-bottom: 60px;
      padding-top: 40px;
    }
  }

  &-divider {
    color: @dte-content-primary-inverse;
    @media @mobile {
      display: none;
    }
  }

  &-links {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    font-size: @dte-wismo-font-size-s;
    justify-content: space-between;
    min-width: 380px;

    @media @mobile {
      align-items: center;
      flex-direction: column;
      height: inherit;
      justify-content: inherit;
      width: inherit;
    }

    > a {
      display: flex;
      flex-direction: row;

      @media @mobile {
        padding-top: 10px;
      }

      &:link,
      &:visited {
        color: @dte-content-primary-inverse !important;
        text-decoration: none;
      }

      &:hover,
      &:active {
        text-decoration: underline;
      }
    }
  }
}
