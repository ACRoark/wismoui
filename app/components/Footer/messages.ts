import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.footer';

export default defineMessages({
  allRightsReserved: {
    id: `${scope}.all-rights-reserved`,
  },
  contactUs: {
    id: `${scope}.contact-us`,
  },
  privacyPolicy: {
    id: `${scope}.privacy-policy`,
  },
  termsAndConditions: {
    id: `${scope}.terms-and-conditions`,
  },
});
