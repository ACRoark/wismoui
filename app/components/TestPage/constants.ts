import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
