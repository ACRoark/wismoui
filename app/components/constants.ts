import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
