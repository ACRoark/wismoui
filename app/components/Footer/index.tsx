import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import useConfig from 'hooks/useConfig';

import './index.less';
import messages from './messages';

const currentYear = new Date().getFullYear();

const Footer: React.FC = (): React.ReactElement => {
  const { urls } = useConfig();

  return (
    <footer className="dte-wismo-footer">
      <div className="dte-wismo-footer-content">
        <div className="dte-wismo-footer-links">
          <a href={urls.privacyPolicy}>
            <FormattedMessage {...messages.privacyPolicy} />
          </a>
          <p className="dte-wismo-footer-divider">|</p>
          <a href={urls.termsAndConditions}>
            <FormattedMessage {...messages.termsAndConditions} />
          </a>
          <p className="dte-wismo-footer-divider">|</p>
          <a href={urls.contactUs}>
            <FormattedMessage {...messages.contactUs} />
          </a>
        </div>
        <div className="dte-wismo-footer-copyright">
          <p>
            <FormattedMessage {...messages.allRightsReserved} values={{ year: currentYear }} />
          </p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
