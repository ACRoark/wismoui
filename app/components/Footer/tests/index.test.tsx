import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import Footer from '..';

jest.mock('hooks/useConfig');

const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;

describe('<Footer />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  const store = createMockStore();

  mockConfig.mockReturnValue({ ...testConfig });

  languages.forEach((language: string): void => {
    it('should not log errors in console when rendered', (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<Provider store={store}><Footer /></Provider>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it('should render correctly', (): void => {
      const tree = createSnapshotWithIntl(<Provider store={store}><Footer /></Provider>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
