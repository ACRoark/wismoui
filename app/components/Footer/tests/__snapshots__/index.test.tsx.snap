// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer
  className="dte-wismo-footer"
>
  <div
    className="dte-wismo-footer-content"
  >
    <div
      className="dte-wismo-footer-links"
    >
      <a
        href="https://www.privacy.policy.com"
      >
        Privacy Policy
      </a>
      <p
        className="dte-wismo-footer-divider"
      >
        |
      </p>
      <a
        href="https://www.terms.and.conditions.com"
      >
        Terms & Conditions
      </a>
      <p
        className="dte-wismo-footer-divider"
      >
        |
      </p>
      <a
        href="https://www.contact.us.com"
      >
        Contact Us
      </a>
    </div>
    <div
      className="dte-wismo-footer-copyright"
    >
      <p>
        © 2021 DTE Energy. All rights reserved
      </p>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should render correctly 2`] = `
<footer
  className="dte-wismo-footer"
>
  <div
    className="dte-wismo-footer-content"
  >
    <div
      className="dte-wismo-footer-links"
    >
      <a
        href="https://www.privacy.policy.com"
      >
        Política de privacidad
      </a>
      <p
        className="dte-wismo-footer-divider"
      >
        |
      </p>
      <a
        href="https://www.terms.and.conditions.com"
      >
        Términos y condiciones
      </a>
      <p
        className="dte-wismo-footer-divider"
      >
        |
      </p>
      <a
        href="https://www.contact.us.com"
      >
        Contacta con nosotras
      </a>
    </div>
    <div
      className="dte-wismo-footer-copyright"
    >
      <p>
        © 2021 DTE Energía. Todos los derechos reservados
      </p>
    </div>
  </div>
</footer>
`;
